src/example-types/metricGroups/dateIntervalGroupStats.js

```diff
@@ -9,16 +12,141 @@ let drilldown = ({ field, interval, drilldown }) => {
     .format()
   return { range: { [field]: { gte, lte } } }
 }
-let buildGroupQuery = ({ field, interval = 'year' }, children, groupsKey) => ({
-  aggs: {
-    [groupsKey]: {
-      date_histogram: { field, interval, min_doc_count: 0 },
-      ...children,
+
+let supportedFederalFiscalYearMetrics = [
+  'sum',
+  // TODO
+  // 'avg',
+  // 'min',
+  // 'max',
+  // 'percentiles', // maybe can do?
+  // 'geo_bounds', // likely can do?
+  // 'geo_centroid', // maybe can do but looks tricky
+]
+
+let isOffsetDates = node => {
+  if (node.interval !== 'federalFiscalYear') return false
+  if (_.isEmpty(stats)) return false
+  return !node.stats.every(val =>
+    supportedFederalFiscalYearMetrics.includes(val)
+  )
+}
+
+let buildGroupQuery = (node, children, groupsKey) => {
+  let { field, interval = 'year' } = node
+
+  // Federal fiscal year quarters have the
+  // same start and end dates as conventional
+  // calendar year quarters but offset forward by one.
+  // e.g. calendarYear2022Q1 => federalFiscalYear2022Q2
+  //      calendarYear2022Q4 => federalFiscalYear2023Q1
+  // In either federalFiscal interval senario,
+  // elasitc computes a date_histogram using `'quarter'`
+  // and the date offset and
+  // year bucketing (in the case of federalFiscalYear)
+  // are computed in node in the `result` function.
+  if (interval === 'federalFiscalQuarter') interval = 'quarter'
+
+  let offsetDates = isOffsetDates(node)
+  if (interval === 'federalFiscalYear') {
+    if (offsetDates) {
+      interval = 'quarter'
+    } else {
+      interval = 'year'
+    }
+  }
+
+  return {
+    ...(offsetDates
+      ? {
+          runtime_mappings: {
+            [`${field}-offset`]: {
+              type: 'date',
+              script: `
+            long date = doc['${field}'].value.toInstant().toEpochMilli();
+            // Both third and forth quarter of a standard year are 92 days
+            // This allows us to simple add 92 days to a date to offset
+            // the dates to correspond to a federal fiscal year.
+            date += 7862400000L; // 92 days
```

Comment:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/ZonedDateTime.html#plusMonths(long)

Reply:
I'm hesitant to go that route because a month is variable length. Adding 92days exactly gives us exactly what we want regardless of the date, where as adding three months may have some unforeseen edge cases (say adding three months to the 31st of a month, what happens if the the month doesn't have a 31st).  This approach is probably more readable but harder to reason about likely has an edge case we won't think of.

Reply:
92 days might work for Q3 and Q4, but I don't think that holds true for other quarters (which might cause bugs for the other cases). Let's do a test to see how it handles the 31st case (e.g. does may 31 end up in q3 or q4?)

Reply:
makes sense, updated.